src/Core/Grid/Column/ColumnCollection.php

```diff
@@ -90,4 +77,34 @@ public function remove($id)
 
         return $this;
     }
+
+    /**
+     * Insert new column into collection at given position
+     *
+     * @param string          $id        Existing column id
+     * @param ColumnInterface $newColumn Column to insert
+     * @param string          $position  Position: "before" or "after"
+     *
+     * @throws ColumnNotFoundException When column with gieven $id does not exist
+     */
+    private function insertByPosition($id, ColumnInterface $newColumn, $position)
```

Comment:
I'd like to see unit tests on that: they're easy to create and will cover one of the most used use cases of the Grid component.

Reply:
you'll save us a lot of technical debt doing this, believe me <3